internal/provider/host_resource.go

```diff
@@ -0,0 +1,554 @@
+		resp.Diagnostics.Append(data.Groups.ElementsAs(ctx, &elements, false)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+		resp.Diagnostics.Append(r.AssociateGroups(ctx, elements, data.URL.ValueString()+pathGroup)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+
+		groups, diags := r.ReadAssociatedGroups(data)
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+
+		resp.Diagnostics.Append(data.UpdateStateWithGroups(ctx, groups)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read refreshes the Terraform state with the latest host data.
+func (r *HostResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data HostResourceModel
+	var diags diag.Diagnostics
+
+	// Read current Terraform state data into host resource model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get latest host data from AAP
+	readResponseBody, diags := r.client.Get(data.URL.ValueString())
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save latest host data into host resource model
+	diags = data.ParseHttpResponse(readResponseBody)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	groups, diags := r.ReadAssociatedGroups(data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(data.UpdateStateWithGroups(ctx, groups)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update updates the host resource and sets the updated Terraform state on success.
+func (r *HostResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data HostResourceModel
+	var diags diag.Diagnostics
+
+	// Read Terraform plan data into host resource model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// create request body from host data
+	updateRequestBody, diags := data.CreateRequestBody()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	requestData := bytes.NewReader(updateRequestBody)
+
+	// Update host in AAP
+	updateResponseBody, diags := r.client.Update(data.URL.ValueString(), requestData)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save updated host data into host resource model
+	diags = data.ParseHttpResponse(updateResponseBody)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if !data.Groups.IsNull() {
```

Comment:
If I have a host that's a member of one or more groups, then I change it to remove all group membership, it doesn't actually update the group membership in AAP. I think you may just need to remove the conditional on this block.

Reply:
To remove all groups memberships you just need to omit groups using the latest changes.